### PR TITLE
DOC: Renamed DHO to IDF

### LINKED ISSUE #1810: Rename DHO to IDF
As of IntelMQ 3.0.0 the internal data format changes from "IntelMQ Data Harmonization" to "IntelMQ Data Format".

intelmq/docs/dev/guide.rst
```
**Reference:** IntelMQ Data Harmonization - :doc:`data-harmonization`
* Message: Defines Events and Reports classes, uses harmonization to check validity of keys and values according to config.
```

intelmq/docs/dev/data-harmonization.rst
```
All messages (reports and events) are Python/JSON dictionaries. The key names and according types are defined by the so called *harmonization*.
Data which does not fit in the harmonization can be saved in the 'extra' namespace. All keys must begin with `extra.`, there are no other rules on key names and values. The values can be get/set like all other fields.
.. _data harmonization classification:
```

intelmq/docs/user/ecosystem.rst:
```
It includes all the bots, the harmonization, etc.
```

intelmq/docs/user/bots.rst
```
* `"columns"`: A list of strings or a string of comma-separated values with field names. The names must match the harmonization's field names. Empty column specifications and columns named `"__IGNORE__"` are ignored. E.g.
* `"columns"`: A list of strings or a string of comma-separated values with field names. The names must match the harmonization's field names. Empty column specifications and columns named `"__IGNORE__"` are ignored. E.g.
* `classification_type`: string with a valid classification type as defined in data harmonization
* `filter_key` - key from data harmonization
* `strip_columns` -  A list of strings or a string of comma-separated values with field names. The names must match the harmonization's field names. E.g.
* `replace_column` - key from data harmonization
* `split_column` - key from data harmonization
from `harmonization.conf`. The script will create the required table layout
```

intelmq/docs/user/configuration-management.rst
```
This configuration is used to specify the fields for all message types. The harmonization library will load this configuration to check, during the message processing, if the values are compliant to the "harmonization" format. Usually, this configuration doesn't need any change. It is mostly maintained by the intelmq maintainers.
More examples can be found in the `intelmq/etc/harmonization.conf` directory.
```

intelmq/docs/user/intelmqctl.rst
```
Note, if you would like to know possible parameters of the message, put a wrong one – you will be prompted if you want to list all the current bot harmonization.
```

intelmq/docs/index.rst
```
dev/data-harmonization
dev/harmonization-fields.md
```

### PR DESCRIPTION
**Pull request by waldbauer-certat**
Renaming IntelMQ Data Harmonization to IntelMQ Data Format. Internal files will remain and work the same as before.

Closes #1810

### CODE REVIEW

File: docs/dev/data-format.rst [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+Data Format
 ##################
 
 .. contents::
 
 Overview
 ========
 
-All messages (reports and events) are Python/JSON dictionaries. The key names and according types are defined by the so called *harmonization*.
+In IntelMQ version 3.x+ the internal data format changed from DHO ( IntelMQ Data Harmonization ) to IDF ( IntelMQ Data Format ).
```
Comment (wagner-certat) [inline docs/dev/data-format.rst:10]: The *name* changed ("... the internal data format name changed ..." or similar)

File: docs/dev/data-format.rst [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+The python module and the configuration file kept the name `harmonization`. DHO and IDF are working similar.
```
Comment (wagner-certat) [inline docs/dev/data-format.rst:11]: please name the module/file explicitly.

> The python module ``intelmq.lib.harmonization`` and the configuration file ``harmonization.conf`` keep the name `harmonization` for now. DHO and IDF have the same meaning.

**Review by wagner-certat — CHANGES_REQUESTED** [review]
We should get rid of the trailing whitespaces of course, but I'd prefer a separate commit as there are a lot of them. (should be reasonable easy to separate with git reset and git add -p).

Please add an entry to the changelog and the news file as well.

Looks very good.

**Review by wagner-certat — APPROVED** [review]

**Review by schacht-certat — APPROVED** [review]

**Comment by wagner-certat** [comment]
Merged in 5ec55cad1e0d52c191886b8add780e6de7e238c2

**Comment by wagner-certat** [comment]
Looks like there was a change to the first commit after the merge. Cherry picked that now with e691090403c1d2b80d1b5ddb46f92735ef9aa75c.

Strangely, I can't find the commit with the whitespace fix, but `git cp c2653` says, the pick would be empty, so I guess the change is in develop?

**Pull request closed without merging**